src/averageAges.js

```diff
@@ -54,11 +69,25 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people
+    .filter(person => people.find(mother => mother.name === person.mother)
+    && (onlyWithSon ? person.sex === 'm' : person));
```

Comment:
```suggestion
    .filter(person => people.some(mother => mother.name === person.mother)
    && (onlyWithSon ? person.sex === 'm' : person));
```
better to use `some` because you just want to know if this item exists in the array